andela/ah-frontend-odin
# #161340152 Add Search Functionality and Custom filtering based on parameters (UI)

#### What does this PR do?
This PR enables users to filter dozens of articles to a few using parameters
#### Description of Task to be completed?
- Filter articles using title, tag, body, and author
#### How should this be manually tested?
- Refer to the README.md for instructions on how to get up and running with this application.
- With the application served and running in the browser, click on the search icon in the navigation bar which will redirect you to a search page.
- Add a **`title`** keyword in the Navigation search input, hit enter and it should return all articles that match the criteria.
- You can also  specify more filters in the filter toolbar like **`author`**, **`body`** and  **`tag`**, and it will return the articles that meet the criteria when you hit apply
#### What are the relevant pivotal tracker stories?
[#161340152](https://www.pivotaltracker.com/story/show/161340152)
#### Screenshots
![search-1](https://user-images.githubusercontent.com/33207024/50318051-f2d07580-04cf-11e9-8684-fdb9dc6a3135.png)
![search-2](https://user-images.githubusercontent.com/33207024/50318056-f9f78380-04cf-11e9-965a-8a207d0da49d.png)
![search-3](https://user-images.githubusercontent.com/33207024/50318058-febc3780-04cf-11e9-8d33-47916311d4f4.png)
![search-4](https://user-images.githubusercontent.com/33207024/50318067-067bdc00-04d0-11e9-975e-d29742097e30.png)
